ui-spacetimechart/src/components/PathLayer.tsx

```diff
@@ -211,10 +211,12 @@ export const PathLayer = ({
     ) => {
       if (!label) return;
 
-      const firstPointOnScreenIndex = points.findIndex(({ x, y }) =>
-        !swapAxis
-          ? inRange(x, 0, width) && inRange(y, 0, height - CAPTION_SIZE)
-          : inRange(x, CAPTION_SIZE, width) && inRange(y, 0, height)
+      const minX = swapAxis ? captionSize : 0;
+      const minY = 0;
+      const maxX = width;
+      const maxY = swapAxis ? height : height - captionSize;
+      const firstPointOnScreenIndex = points.findIndex(
+        ({ x, y }) => inRange(x, minX, maxX) && inRange(y, minY, maxY)
```

Comment:
Hum I found the previous version of the code clearer
```suggestion
    const firstPointOnScreenIndex = points.findIndex(({ x, y }) =>
        !swapAxis
          ? inRange(x, 0, width) && inRange(y, 0, height - captionSize)
          : inRange(x, captionSize, width) && inRange(y, 0, height)
```

Also, if you get back to the first version of the code, I think you can drop the maxX and maxY

Reply:
Agreed. I just pushed a fix for that. Thanks for that feedback :)